usaspending_api/references/v2/views/naics.py

```diff
@@ -0,0 +1,178 @@
+            tier2_codes.add(naic.code[:4])
+            tier1_codes.add(naic.code[:2])
+        tier2_naics = NAICS.objects.annotate(text_len=Length("code")).filter(**naics_filter, text_len=4)
+
+        for naic in tier2_naics:
+            tier2_codes.add(naic.code)
+            tier1_codes.add(naic.code[:2])
+        extra_tier2_naics = NAICS.objects.annotate(text_len=Length("code")).filter(code__in=tier2_codes, text_len=4)
+        tier1_naics = NAICS.objects.annotate(text_len=Length("code")).filter(text_len=2, code__in=tier1_codes)
+
+        tier2 = set(list(tier2_naics))
+        ex2 = list(extra_tier2_naics)
+        for naic in ex2:
+            tier2.add(naic)
+        tier2_results = {}
+        for naic in tier2:
+            result = OrderedDict()
+            result["naics"] = naic.code
+            result["naics_description"] = naic.description
+            result["count"] = (
+                NAICS.objects.annotate(text_len=Length("code")).filter(code__startswith=naic.code, text_len=6).count()
+            )
+            result["children"] = []
+            tier2_results[naic.code] = result
+        for naic in tier3_naics:
+            result = OrderedDict()
+            result["naics"] = naic.code
+            result["naics_description"] = naic.description
+            result["count"] = 0
+            tier2_results[naic.code[:4]]["children"].append(result)
+        tier1_results = {}
+        for naic in tier1_naics:
+            result = OrderedDict()
+            result["naics"] = naic.code
+            result["naics_description"] = naic.description
+            result["count"] = (
+                NAICS.objects.annotate(text_len=Length("code")).filter(code__startswith=naic.code, text_len=6).count()
+            )
+            result["children"] = []
+            tier1_results[naic.code] = result
+        for key in tier2_results.keys():
+            tier1_results[key[:2]]["children"].append(tier2_results[key])
+        results = []
+        for key in tier1_results.keys():
+            results.append(tier1_results[key])
+        response_content = OrderedDict({"results": results})
+        return response_content
+
+    def _default_view(self) -> dict:
+        naics = NAICS.objects.annotate(text_len=Length("code")).filter(text_len=2)
+        results = []
+        for naic in naics:
+            result = OrderedDict()
+            result["naics"] = naic.code
+            result["naics_description"] = naic.description
+            result["count"] = (
+                NAICS.objects.annotate(text_len=Length("code")).filter(code__startswith=naic.code, text_len=6).count()
+            )
+            results.append(result)
+        response_content = OrderedDict({"results": results})
+        return response_content
+
+    def _business_logic(self, request_data: dict) -> dict:
+        naics_filter = {}
+        code = request_data.get("code")
+        description = request_data.get("description")
```

Comment:
wasn't this called filter in the readme?

Reply:
yeah, but python already has a built in called `filter` so we don't want it to shadow that. we also assigned this to be called 'description' in parse_and_validate_request.